Remove user from list of coteachers on teacher reg mainpage

willgearty opened the pull request:
This makes it so teachers no longer show up as their own coteachers on the teacher registration mainpage:
![image](https://user-images.githubusercontent.com/7232514/142094708-64af8f7a-92f2-49c4-a97f-465a4dbf7b10.png)

Fixes #3261.

kkbrum commented on esp/templates/inclusion/program/class_teacher_list_row.html:
```diff
@@ -108,8 +108,8 @@ <h3>Cancellation Request for {{ cls|truncatewords:6 }}</h3>
-  <td class="clsleft"><b>Teachers (<a href="mailto:{{ cls.emailcode }}-teachers@{{ email_host_sender }}">email</a>): </b></td>
-  <td class="clsmiddle" colspan="5">{% for t in cls.get_teachers %}{% ifnotequal t request.user %}{{ t.first_name }} {{ t.last_name }}{% if not forloop.last %}, {% endif %}{% endifnotequal %}{% endfor %}</td>
+  <td class="clsleft"><b>Coteachers (<a href="mailto:{{ cls.emailcode }}-teachers@{{ email_host_sender }}">email</a>): </b></td>
+  <td class="clsmiddle" colspan="5">{% for t in cls.get_teachers %}{% ifnotequal t user %}{{ t.first_name }} {{ t.last_name }}{% if not forloop.last %}, {% endif %}{% endifnotequal %}{% endfor %}</td>
```
The multiple if's in this line generate a slight problem. If it's not the last teacher, it puts a comma, but then when it goes to the next teacher and it's equal to the user, it passes. So if the user happens to be last in the list then you end up with a trailing comma

willgearty commented:
I've added a template filter that removes the teacher from the list, so now there shouldn't be any extra commas. I did the same thing for the teacher webapp sectioninfo page (and cleaned up the moderators section to match the teacher registration mainpage).

kkbrum reviewed (approved):
Nice!
